source/jormungandr/jormungandr/scenarios/experimental.py

```diff
@@ -244,90 +400,89 @@ def call_kraken(self, request_type, request, instance, krakens_call):
         logger = logging.getLogger(__name__)
         logger.debug('datetime: %s', request['datetime'])
         crow_fly_stop_points = set()
+
         if not g.requested_origin:
             g.requested_origin = instance.georef.place(request['origin'])
         if not g.requested_destination:
             g.requested_destination = instance.georef.place(request['destination'])
 
-        for dep_mode, arr_mode in krakens_call:
-            if dep_mode not in g.origins_fallback and request.get('max_duration', 0):
-                g.origins_fallback[dep_mode] = self._get_stop_points(instance,
-                                                                     g.requested_origin, dep_mode,
-                                                                     get_max_fallback_duration(request, dep_mode),
-                                                                     request)
-
-                #Fetch all the stop points of this stop_area and replaces all the durations by 0 in the table
-                #g.origins_fallback[dep_mode]
-                _update_crowfly_duration(instance, g.origins_fallback, dep_mode, request['origin'],
-                                         crow_fly_stop_points)
-
-            if arr_mode not in g.destinations_fallback and request.get('max_duration', 0):
-               g.destinations_fallback[arr_mode] = self._get_stop_points(instance,
-                                                                         g.requested_destination, arr_mode,
-                                                                         get_max_fallback_duration(request, arr_mode),
-                                                                         request,
-                                                                         reverse=True)
-
-               #Fetch all the stop points of this stop_area and replaces all the durations by 0 in the table
-               #g.destinations_fallback[arr_mode]
-               _update_crowfly_duration(instance, g.destinations_fallback, arr_mode, request['destination'],
-                                        crow_fly_stop_points)
+        worker = AsyncWorker(instance, krakens_call, request)
+
+        if request.get('max_duration', 0):
+            orig_futures, dest_futures = worker.get_crowfly_futures(g.requested_origin, g.requested_destination)
+            gevent.joinall(orig_futures + dest_futures)
+            for future in gevent.iwait(orig_futures):
```

Comment:
I imagined a scenario as followed: 

We need to compute some crowfly stop_points with "walking" and "bike" respectively for "from" and "to", in this case we'll get 4 requests for stop_points.

**WITH** `joinall`: all requests are launched at the same time thanks to the concatenation
**WITHOUT** `joinall`: only 2 requests are parallelized 

That's why i did with `joinall`